main/src/main/scala/sbt/RemoteCache.scala

```diff
@@ -113,6 +113,11 @@ object RemoteCache {
         DiskActionCacheStore(localCacheDirectory.value.toPath(), fileConverter.value)
       )
     },
+    remoteCache := SysProp.remoteCache,
+    remoteCacheTlsCertificate := SysProp.remoteCacheTlsCertificate,
+    remoteCacheTlsClientCertificate := SysProp.remoteCacheTlsClientCertificate,
+    remoteCacheTlsClientKey := SysProp.remoteCacheTlsClientKey,
+    remoteCacheHeaders := SysProp.remoteCacheHeaders,
```

Comment:
In general I think we should be careful in adding default settings and tasks. Loading big project matrices is already slow and adding more settings will make it slower.

Should we move those setting to the `RemoteCachePlugin`?

Reply:
I think the ones that we have to be careful is any keys that gets added at project level, like many of the plugins, like sbt-dependency-graph, which I think adds about a dozen keys, and in projects with 100 subprojects that ends up adding to initialization time. Adding a few hardcoded values at the global level is I think is almost free in that sense. I should change to `:==` which skips graph evaluation.

> Should we move those setting to the `RemoteCachePlugin`?

Here are some reasons to keep it in mothership:
1. Other plugins that want to implement remote backend can reuse this key without depending on sbt-remote-cache.
2. Build users can write the settings in `build.sbt`, but during the build time selectively ignore sbt-remote-cache (for example, in-house usage vs public).

Reply:
> I think the ones that we have to be careful is any keys that gets added at project level

Oh yes you are absolutely right